### PR TITLE
Update xunit to v2.6.2

### PR DESCRIPTION
**Pull request by ViktorHofer**
**Important: MERGE COMMIT**

FWIW: I used a commit instead of a tag because of https://github.com/xunit/xunit/issues/2843

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Triggering the `dotnet-public` sync for the following package ids: `xunit.runner.reporters,xunit.runner.utility,xunit.runner.console`

**Comment by ViktorHofer** [comment]
Yeah another change that breaks AOT compatibility:

> src\Microsoft.DotNet.XUnitAssert\src\Sdk\AssertEqualityComparer.cs(47,24): error IL3050: (NETCORE_ENGINEERING_TELEMETRY=Build) Using member 'System.Type.MakeGenericType(params Type[])' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. The native code for this instantiation might not be available at runtime.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props [hunk 1]
```diff
@@ -81,8 +81,8 @@
     <MicrosoftDotNetSignToolVersion Condition="'$(MicrosoftDotNetSignToolVersion)' == ''">$(ArcadeSdkVersion)</MicrosoftDotNetSignToolVersion>
     <MicrosoftDotNetTarVersion Condition="'$(MicrosoftDotNetTarVersion)' == ''">$(ArcadeSdkVersion)</MicrosoftDotNetTarVersion>
     <MicrosoftTestPlatformVersion Condition="'$(MicrosoftTestPlatformVersion)' == ''">16.5.0</MicrosoftTestPlatformVersion>
-    <XUnitVersion Condition="'$(XUnitVersion)' == ''">2.6.1</XUnitVersion>
-    <XUnitAnalyzersVersion Condition="'$(XUnitAnalyzersVersion)' == ''">1.4.0</XUnitAnalyzersVersion>
+    <XUnitVersion Condition="'$(XUnitVersion)' == ''">2.6.2</XUnitVersion>
+    <XUnitAnalyzersVersion Condition="'$(XUnitAnalyzersVersion)' == ''">1.6.0</XUnitAnalyzersVersion>
     <XUnitRunnerConsoleVersion Condition="'$(XUnitRunnerConsoleVersion)' == ''">$(XUnitVersion)</XUnitRunnerConsoleVersion>
     <XUnitRunnerVisualStudioVersion Condition="'$(XUnitRunnerVisualStudioVersion)' == ''">$(XUnitVersion)</XUnitRunnerVisualStudioVersion>
```
Comment (mmitche) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props:87]: There's no 2.6.2 version of XRV available on nuget.org

Comment (ViktorHofer) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: Thanks, fixed.

**Pull request closed without merging**